CODE_OF_CONDUCT.md

```diff
@@ -0,0 +1,128 @@
+and orientation.
+
+We pledge to act and interact in ways that contribute to an open, welcoming,
+diverse, inclusive, and healthy community.
+
+## Our Standards
+
+Examples of behavior that contributes to a positive environment for our
+community include:
+
+* Demonstrating empathy and kindness toward other people
+* Being respectful of differing opinions, viewpoints, and experiences
+* Giving and gracefully accepting constructive feedback
+* Accepting responsibility and apologizing to those affected by our mistakes,
+  and learning from the experience
+* Focusing on what is best not just for us as individuals, but for the
+  overall community
+
+Examples of unacceptable behavior include:
+
+* The use of sexualized language or imagery, and sexual attention or
+  advances of any kind
+* Trolling, insulting or derogatory comments, and personal or political attacks
+* Public or private harassment
+* Publishing others' private information, such as a physical or email
+  address, without their explicit permission
+* Other conduct which could reasonably be considered inappropriate in a
+  professional setting
+
+## Enforcement Responsibilities
+
+Community leaders are responsible for clarifying and enforcing our standards of
+acceptable behavior and will take appropriate and fair corrective action in
+response to any behavior that they deem inappropriate, threatening, offensive,
+or harmful.
+
+Community leaders have the right and responsibility to remove, edit, or reject
+comments, commits, code, wiki edits, issues, and other contributions that are
+not aligned to this Code of Conduct, and will communicate reasons for moderation
+decisions when appropriate.
+
+## Scope
+
+This Code of Conduct applies within all community spaces, and also applies when
+an individual is officially representing the community in public spaces.
+Examples of representing our community include using an official e-mail address,
+posting via an official social media account, or acting as an appointed
+representative at an online or offline event.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported to the community leaders responsible for enforcement at
+contact@diasurgical.org.
+All complaints will be reviewed and investigated promptly and fairly.
+
+All community leaders are obligated to respect the privacy and security of the
+reporter of any incident.
+
+## Enforcement Guidelines
+
+Community leaders will follow these Community Impact Guidelines in determining
+the consequences for any action they deem in violation of this Code of Conduct:
+
+### 1. Correction
+
+**Community Impact**: Use of inappropriate language or other behavior deemed
+unprofessional or unwelcome in the community.
+
+**Consequence**: A private, written warning from community leaders, providing
+clarity around the nature of the violation and an explanation of why the
+behavior was inappropriate. A public apology may be requested.
```

Comment:
I'd leave public apologies out of it.

Personally apologizing to the offended party is what reasonable people do. Public apologies, if not done of one's own initiative, are a form of humiliation. I don't think we should be in the business of humiliating offenders.

Reply:
I agree. If a public apology is deemed necessary or useful by the enforcer, then the situation should be explained to the offending party in a way that encourages them to make their own decision about whether to make the apology. In that way, I see it as more of a strategy for conflict resolution than a matter of enforcement procedure. The phrasing suggests that we won't be demanding any public apology, but its presence in the code of conduct and even its location within the text suggest that it's some kind of consequence or punishment that community members need to be made aware of.

Reply:
a public apology **may** be requested, we don't have to do that, I'd leave it

Reply:
If we were to request a public apology of someone in the community, where would that apology go? I feel that GitHub isn't even the appropriate venue for such a public statement in any realistic situation. Furthermore, I feel like these guidelines don't even address the types of enforcement we typically take in response to conflict, such as deleting off-topic comments in heated discussions and temporarily/permanently locking issues to prevent further discussion.

> Before you decide to adopt Contributor Covenant, take the time to discuss enforcement with other trusted members of your community. Thoroughly read the Enforcement Guidelines section of the document. If the suggested actions and consequences don’t suit the context of your community, consider moving this section to another document (such as an FAQ or ‘How to Report’ page) and adapting it to your needs.

I feel this section probably needs a significant adjustment.

Reply:
I agree that requesting a public apology is both extreme and likely ineffective. But like Qndel pointed out it says "may", and even then it would be a suggestion and not a requirement.

Unless there is something egregious I would prefer that we keep the CoC unmodified. The default version should work in general as it is [adopted by a large number of projects](https://www.contributor-covenant.org/adopters/), and making changes risks us making bad calls and spending a lot of time doing so.

Also keep in mind that a CoC applies to our community, which also includes the Discord chat server and other forum we may provide, not just GitHub.

Reply:
Technically, you *may* request anything. Requesting a public apology was singled out for inclusion, so it is a threat of some sort. I don't think it would go down well, even if the offender acknowledges and owns up to his/her wrongdoing, and is repentant. Imagine being asked to publicly humiliate yourself for the privilege of being allowed to do more unpaid development work.

If I understand correctly, a lot of other projects use this stipulation that is "extreme and likely ineffective", therefore, we should as well.... come on!

Reply:
> Unless there is something egregious I would prefer that we keep the CoC unmodified. The default version should work in general as it is [adopted by a large number of projects](https://www.contributor-covenant.org/adopters/), and making changes risks us making bad calls and spending a lot of time doing so.

That's something I don't really agree with. The excerpt I quoted about discussing with the community and discussing the Enforcement Guidelines comes straight from the Contributor Covenant main page. Furthermore, a random sampling of projects on the list seems to indicate that the vast majority of those projects are not using this Enforcement Guidelines section. Granted, this could be partially due to the fact that many of them used an earlier version of the CoC, but the point is that the number of users is not indicative of whether we should leave it as-is. Furthermore, many of them include a note at the bottom indicating that it was *adapted* from a specific version of the Contributor Covenant. I really think we should be discussing changing this one section at least.